Classes/Projection/Doctrine/AbstractDoctrineProjector.php

```diff
@@ -105,18 +122,64 @@ protected function remove($object)
      * @return void
      * @api
      */
-   public function reset()
-   {
-       $this->projectionPersistenceManager->drop($this->getReadModelClassName());
-   }
+    public function reset()
+    {
+        $this->projectionPersistenceManager->drop($this->getReadModelClassName());
+    }
 
-   /**
-    * If this projection is currently empty
-    *
-    * @return bool
-    */
-   public function isEmpty()
-   {
-       return $this->projectionPersistenceManager->count($this->getReadModelClassName()) === 0;
-   }
+    /**
+     * If this projection is currently empty
+     *
+     * @return bool
+     */
+    public function isEmpty()
+    {
+        return $this->projectionPersistenceManager->count($this->getReadModelClassName()) === 0;
+    }
+
+    /**
+     * Returns the last seen sequence number of events which has been applied to the concrete event listener.
+     *
+     * @return int
+     */
+    public function getHighestAppliedSequenceNumber(): int
+    {
+        $projectorIdentifier = $this->renderProjectorIdentifier(get_class($this));
```

Comment:
Isn't it a bit weird to pass the projection identifier to the projector? Theoretically you could then tell the `ProjectProjector` to save the sequence number for the `CustomerProjector`.

My reasoning was to make this identifier an implementation detail of the `AbstractDoctrineProjector` - nobody outside that projector should care how the sequence number is stored. But of course we could use the same syntax to make it more consistent.